Machine Learning/Analytics Jupyter Notebooks/ml-analytics-classification.ipynb

```diff
@@ -0,0 +1,469 @@
+       "     'OriginCountry': 'DE',\n",
+       "     'dayOfWeek': 0,\n",
+       "     'DistanceKilometers': 16492.32665375846,\n",
+       "     'timestamp': '2019-11-25T00:00:00',\n",
+       "     'DestLocation': {'lat': '-33.94609833', 'lon': '151.177002'},\n",
+       "     'DestAirportID': 'SYD',\n",
+       "     'Carrier': 'Kibana Airlines',\n",
+       "     'Cancelled': False,\n",
+       "     'FlightTimeMin': 1030.7704158599038,\n",
+       "     'Origin': 'Frankfurt am Main Airport',\n",
+       "     'OriginLocation': {'lat': '50.033333', 'lon': '8.570556'},\n",
+       "     'DestRegion': 'SE-BD',\n",
+       "     'OriginAirportID': 'FRA',\n",
+       "     'OriginRegion': 'DE-HE',\n",
+       "     'DestCityName': 'Sydney',\n",
+       "     'FlightTimeHour': 17.179506930998397,\n",
+       "     'FlightDelayMin': 0}}]}}"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "## insert example of reading docs from ES index\n",
+    "\n",
+    "results = es_client.search(index='kibana_sample_data_flights', filter_path=['hits.hits._*'], size=1)\n",
+    "results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that each document contains a FlightDelay field with a boolean value. Classification is a supervised machine learning analysis and therefore needs to train on data that contains the ground truth, known as the dependent_variable. In this example, the ground truth is available in each document as the actual value of FlightDelay. In order to be analyzed, a document must contain at least one field with a supported data type (numeric, boolean, text, keyword or ip) and must not contain arrays with more than one item.\n",
+    "\n",
+    "If your source data consists of some documents that contain a dependent variable and some that do not, the model is trained on the subset of documents that contain ground truth. By default, all of that subset of documents is used for training. However, you can choose to specify a percentage of the documents as your training data. Predictions are made against all of the data. The current implementation of classification analysis supports a single batch analysis for both training and predictions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Creating a Classification Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'allow_lazy_start': False,\n",
+      " 'analysis': {'classification': {'dependent_variable': 'FlightDelay',\n",
+      "                                 'num_top_classes': 2,\n",
+      "                                 'prediction_field_name': 'FlightDelay_prediction',\n",
+      "                                 'training_percent': 10.0}},\n",
```

Comment:
it might be worth explaining why you've chosen to train on only 10% of the data.

Reply:
and the pros and cons of larger / smaller values